orchestrator/graphql/schemas/product.py

```diff
@@ -0,0 +1,27 @@
+import strawberry
+
+from orchestrator.graphql.schemas.fixed_input import FixedInputType
+from orchestrator.graphql.schemas.product_block import ProductBlockType
+from orchestrator.schemas.product import ProductSchema
+
+
+@strawberry.experimental.pydantic.type(model=ProductSchema)
+class Product:
+    product_id: strawberry.auto
```

Comment:
I did do that initially, but then saw a warning in the logs about that arg (from their code):
```python
def type(
    model: Type[PydanticModel],
    *,
    fields: Optional[List[str]] = None,
    # ...
        if fields:
            warnings.warn(
                "`fields` is deprecated, use `auto` type annotations instead",
                DeprecationWarning,
                stacklevel=2,
            )
```
Not sure if/when they will remove support for `fields`. I agree that defining the fields in the decorator is less verbose and I will convert it back to that style, as well as update the model fields.

Reply:
After experimenting with the `fields` and seeing if strawberry auto-detects the field types defined in other modules, it seems that if I modify `ProductBlockType` to be `ProductBlockSchema`, and register it as such, I get
```
strawberry.experimental.pydantic.exceptions.UnregisteredTypeException: Cannot find a Strawberry Type for <class 'orchestrator.schemas.product_block.ProductBlockSchema'> did you forget to register it?
```
unless I leave the strawberry `ProductBlockSchema` type as an unused import at the top of `product.py`. Is it okay to be explicit about defining the types in this case?

Reply:
It's because I have all of the strawberry types in the same file, I think we need to initialize the strawberry types on startup.

I think this is fine then, will check at somepoint if it can work with initializing the types.